token/wallet.go

```diff
@@ -159,14 +159,20 @@ func (w *Wallet) ContainsToken(token *token2.UnspentToken) bool {
 
 // AuditorWallet models the wallet of an auditor
 type AuditorWallet struct {
-	w api2.AuditorWallet
+	w   api2.AuditorWallet
+	tms *ManagementService
```

Comment:
this change feels wrong to me. Now we want to add a tms reference to a wallet, we need to do this at `AuditorWallet`, `OwnerWallet`, `IssuerWallet`, `MyWallet`, `EmptyWallet`, `YourWallet`, and all the other wallets we may add in the future? This does not scale. Ok, I understand that each wallet type provides special functionality. Why not defining a `Wallet` Struct , which we use to compose our special wallets. The `Wallet` can contain the tms instance, provide the functions for ID, contains, ... just everything that is shared.

Reply:
good point, indeed we have it already. I'll move the field and the method to that common structure